test/internal/layers/layerfolders.go

```diff
@@ -0,0 +1,85 @@
+//go:build windows
+
+package layers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/mount"
+
+	testctrd "github.com/Microsoft/hcsshim/test/internal/containerd"
+)
+
+var imageLayers map[string][]string
+
+func init() {
+	imageLayers = make(map[string][]string)
+}
+
+func FromChainID(ctx context.Context, t testing.TB, client *containerd.Client, chainID, snapshotter string) []string {
+	ms := testctrd.CreateViewSnapshot(ctx, t, client, snapshotter, chainID, chainID+"view")
+	if len(ms) != 1 {
+		t.Fatalf("Rootfs does not contain exactly 1 mount for the root file system")
+	}
+
+	return FromMount(ctx, t, ms[0])
+}
+
+func FromMount(_ context.Context, t testing.TB, m mount.Mount) (layers []string) {
+	for _, option := range m.Options {
+		if strings.HasPrefix(option, mount.ParentLayerPathsFlag) {
+			err := json.Unmarshal([]byte(option[len(mount.ParentLayerPathsFlag):]), &layers)
+			if err != nil {
+				t.Fatalf("failed to unmarshal parent layer paths from mount: %v", err)
+			}
+		}
+	}
+	layers = append(layers, m.Source)
+
+	return layers
+}
+
+func LayerFolders(t testing.TB, imageName string) []string {
+	if _, ok := imageLayers[imageName]; !ok {
+		imageLayers[imageName] = getLayers(t, imageName)
+	}
+	return imageLayers[imageName]
+}
+
+func getLayers(t testing.TB, imageName string) []string {
+	cmd := exec.Command("docker", "inspect", imageName, "-f", `"{{.GraphDriver.Data.dir}}"`)
```

Comment:
I don't understand this. Why are we using docker to get the layers here?

Reply:
We did that in the [older tests](https://github.com/microsoft/hcsshim/blob/master/test/functional/utilities/layerfolders.go#L28) and in [uvmboot](https://github.com/microsoft/hcsshim/blob/master/internal/tools/uvmboot/wcow.go#L125)
I am not sure if theres another way ....